packages/integration-tests/src/entities/Author.test.ts

```diff
@@ -226,6 +226,29 @@ describe("Author", () => {
     expect(rows[0].number_of_books).toEqual(1);
   });
 
+  it("can have async derived tsvector values", async () => {
+    const em = newEntityManager();
+
+    // When we create an author with a first and last name
+    const a1 = new Author(em, { firstName: "Bob", lastName: "Smith" });
+    await em.flush();
+    await em.refresh();
+
+    // a1.search.get is returning "Bob Smith BS" instead of "'bob':1 'bs':3 'smith':2"
+    // expect(a1.search.get).toEqual("Bob Smith BS");
+
+    // Then we expect the firstName, lastName and initials to be cast as tsvector lexemes
+    expect(a1.__orm.data.search).toEqual("'bob':1 'bs':3 'smith':2");
```

Comment:
Not exactly sure the best way to test this - it's an odd scenario where the format is mutated by the database itself so the string in memory in the EM at the time of flush won't match what's returned. 

I needed to both `refresh()` and use `a1.__orm.data.search` vs `a1.search.get` to observe the string of lexemes vs. the input string 🤔

Reply:
Ah huh! yeah, this is interesting.

It breaks the current assumption that Joist has, both fundamentally that:

- A primitive field getter will generally always return the `__orm.data` value--granted we do have FKs where that is not true

And:

- If application code happened to load `Author` row, with `search` _not_ loaded, Joist would duly return the `__orm.data` value, at which point we've leaked the lexemes into the application code, which was probably expect to still see the non-lexemed text (who knows why, but just in theory)

I dunno, this sounds kinda odd, but naively is there a way to do the lexeme-ification in javascript? Like have the derived property return `lexeme("field a field b field c")`?

That would preserve that ".get" at least consistently returns lexemes, so there is not a surprise to the user about it "sometimes returns x/sometimes returns y".

...also, I wouldn't be surprised if we're UPDATE-ing the author, with the current PR would we re-tsvector the value on update? Because we'd have `__orm.data.search` as `robert:1`, b/c that's what we'd loaded from the database, if we hadn't explicitly re-calced/loaded the `search` property's load hint? I.e. `.get` if unloaded will fallback on returning the `robert:1` value...

Granted, if `search` is not a changed field, it shouldn't be included in the `UPDATE`, but if we're updating 3 Authors, and one has its searched changed, and 2 had they're first name changed, we do a single UPDATE with the superset of their fields.

If we did the lexeme in JS, it would also this re-lexeme-ification problem...

Generally I would have assumed that lexem-ification was something really complicated and produced esoteric binary data, but since we're asserting it against it here in the test, it doesn't look too bad, and is maybe something we could calc in JS?

Would the postgres update be okay with that? If the field type was `ts_vector` in the db, and we set it to an already-lexme-ified value, vs. specifically going through `to_tsvector`?

Reply:
https://www.npmjs.com/package/to-tsvector :exploding_head: